crux-core/src/crux/api/v2/CruxId.java

```diff
@@ -0,0 +1,39 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.UUID;
+
+public class CruxId {
+    private final Object id;
+
+    private CruxId(Object id) {
+        this.id = id;
+    }
+
+    public static CruxId cruxId(String name) {
+        return new CruxId(Keyword.intern(name));
+    }
+
+    public static CruxId cruxId(UUID uuid) {
+        return new CruxId(uuid);
+    }
+
+    public static CruxId cruxId(URI uri) {
+        return new CruxId(uri);
+    }
+
+    public static CruxId cruxId(URL url) {
+        return new CruxId(url);
+    }
+
```

Comment:
https://opencrux.com/docs#transactions-valid-ids mentions `Keyword` as a valid id, which is great for meta-model (e.g. store multi-lingual translation of attribute name in the database, track per-attribute permissions, etc) .

Should we also add "type-safe" constructor for `Attribute` as well?
```suggestion

    public static CruxId cruxId(Attribute attribute) {
        return new CruxId(url);
    }
    
```

of course, `Object` already covers all the cases, this overload is only for the documentation / IDE help

Reply:
We were considering removing Attribute/LogicVar and making a Util class from which you can get a Keyword/Symbol, though I do believe making an explicit type safe constructor for Keywords is a good idea.

Reply:
I'd vote for an explicit `Attribute` against very generic `Keyword`

Reply:
reckon we should leave this one open for the time being - there are other places where we'll need Clojure keywords, and need to consider whether we want to expose these as an implementation detail or not (and, indeed, whether we can avoid exposing them as an implementation detail for the time being)